openshift/factstore.app.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: log-anomaly-detector-factstore
+  namespace: aiops-prod-log-analysis
+objects:
+  - apiVersion: image.openshift.io/v1
+    kind: ImageStream
+    metadata:
+      labels:
+        app: log-anomaly-detector-factstore
+      name: log-anomaly-detector-factstore
+    spec:
+      lookupPolicy:
+        local: true
+  - apiVersion: build.openshift.io/v1
+    kind: BuildConfig
+    metadata:
+      labels:
+        app: log-anomaly-detector-factstore
+      name: anomaly-detector-factstore
+    spec:
+      output:
+        to:
+          kind: ImageStreamTag
+          name: log-anomaly-detector-factstore:latest
+      source:
+        git:
+          ref: master
+          uri: https://github.com/AICoE/log-anomaly-detector.git
+        type: Git
+      strategy:
+        sourceStrategy:
+          from:
+            kind: ImageStreamTag
+            name: python:3.6
+            namespace: openshift
+        type: Source
+      triggers:
+        - imageChange: {}
+          type: ImageChange
+        - type: ConfigChange
+  - apiVersion: apps.openshift.io/v1
+    kind: DeploymentConfig
+    metadata:
+      labels:
+        app: log-anomaly-detector-factstore
+      name: log-anomaly-detector-factstore
+    spec:
+      replicas: 1
+      selector:
+        deploymentconfig: log-anomaly-detector-factstore
+      template:
+        metadata:
+          labels:
+            app: log-anomaly-detector-factstore
+            deploymentconfig: log-anomaly-detector-factstore
+        spec:
+          containers:
+            - name: log-anomaly-detector-factstore
+              image: log-anomaly-detector-factstore:latest
+              command: ["python"]
+              args: ["app.py", "ui"]
+              imagePullPolicy: Always
+              resources:
+                limits:
+                  cpu: 2
```

Comment:
I would recommend adding both limits and requests.

Also, 2cpus sounds like a lot, as this is "just" a Flask application, I'd say 500m should be plenty. Or have you arrived at this number by testing? In that case all good:)

Reply:
@zmhassan please use. Afaik the flask app doesnt make use of multiple cores anyway because of GIL

```yaml
limits:
  cpu: 1
  memory: 2Gi
requests:
  cpu: 500m
  memory: 500Mi
```

Reply:
Agreed. I'll update the limits.

Reply:
Done